src/integrator_interface.jl

```diff
@@ -400,6 +400,7 @@ function Base.iterate(integrator::DEIntegrator,state=0)
 end
 
 Base.eltype(integrator::DEIntegrator) = typeof(integrator)
```

Comment:
BTW, while you're at it, `eltype` should be defined for types:
```suggestion
Base.eltype(::Type{T}) where {T<:DEIntegrator} = T
```
I'm also wondering how useful (and reasonable) this definition is at all?